pyomo/solvers/plugins/solvers/mosek_direct.py

```
@@ -163,21 +163,25 @@ def _process_stream(msg):
         for key, option in self.options.items():
             try:
                 param = key.split('.')
-                if param[0] == 'mosek':
-                    param.pop(0)
-                param = getattr(mosek, param[0])(param[1])
-                if 'sparam' in key.split('.'):
-                    self._solver_model.putstrparam(param, option)
-                elif 'dparam' in key.split('.'):
-                    self._solver_model.putdouparam(param, option)
-                elif 'iparam' in key.split('.'):
-                    if isinstance(option, str):
-                        option = option.split('.')
-                        if option[0] == 'mosek':
-                            option.pop('mosek')
-                        option = getattr(mosek, option[0])(option[1])
-                    else:
-                        self._solver_model.putintparam(param, option)
+                if key == param[0]:
+                    self._solver_model.putparam(key, option)
+                else:
+                    if param[0] == 'mosek':
+                        param.pop(0)
+                    param = getattr(mosek, param[0])(param[1])
+                    if 'sparam' in key.split('.'):
+                        self._solver_model.putstrparam(param, option)
+                    elif 'dparam' in key.split('.'):
+                        self._solver_model.putdouparam(param, option)
+                    elif 'iparam' in key.split('.'):
```

I worry that there is no `else:` clause: users who forget to explicitly include `.sparam`, `.dparam`, or `.iparam` will have those options silently ignored.

---

Before the `if` clause, I call `param = getattr(mosek, param[0])(param[1])`. If an invalid key is in the options dict, it should result in an `AttributeError`. So I thought it should be ok even without an else statement.

---

Actually, I think the "else"  is still worthwhile.  As written, if a user provides a key `mosek.mark.up`, nothing will happen and there will be no warning/error.  Maybe not a big deal (it took me a bit of digging to actually come up with this...)?

What would you think of something like this:
```python
if key.startswith('mosek.'):
    key = key.split('.', 1)[1]
param = key.split('.', 1)
if len(param) == 1:
    self._solver_model.putparam(key, option)
else:
    param_obj = getattr(mosek, param[0])(param[1])
    if param[0] == 'sparam':
        self._solver_model.putstrparam(param_obj, option)
    elif param[0] == 'dparam':
        self._solver_model.putdouparam(param_obj, option)
    elif param[0] == 'iparam':
        if isinstance(option, str):
            if option.startswith('mosek'.):
                option = option.split('.', 1)[1]
            option = option.split('.', 1)
            option = getattr(mosek, option[0])(option[1])
        self._solver_model.putintparam(param_obj, option)
    else:
        raise ValueError(f"unrecognized option name '{key}'")
```

---

You make a good point, I will make the change. Thank you for the suggestion.